provider/docker/docker.go

```diff
@@ -711,6 +713,31 @@ func (p *Provider) getBasicAuth(container dockerData) []string {
 	return []string{}
 }
 
+func (p *Provider) getRequestHeaders(container dockerData) map[string]string {
+	h := make(map[string]string)
+	if requestHeaders, err := getLabel(container, types.LabelFrontendRequestHeader); err == nil {
+		for _, v := range strings.Split(requestHeaders, ",") {
+			k := strings.Split(v, ":")
+			h[k[0]] = k[1]
+		}
+	}
+	log.Debugf("Loaded Request Headers: %v", h)
+	return h
+}
+
+func (p *Provider) getResponseHeaders(container dockerData) map[string]string {
+	h := make(map[string]string)
+	if responseHeaders, err := getLabel(container, types.LabelFrontendResponseHeader); err == nil {
+		for _, v := range strings.Split(responseHeaders, ",") {
+			k := strings.Split(v, ":")
+			h[k[0]] = k[1]
+		}
+	}
```

Comment:
The parse logic for the requests and responses seems identical. Might be worth extracting into a little helper function.

Reply:
Seems legit.